test/runtests.jl

```diff
@@ -76,6 +76,14 @@ let
     @test y == 1
 end
 
+# Slurping (parametric case)
+
+@qstruct SlurpParam{T}(x::AbstractVector{T}, y=1, args...; kwargs...)
+s = SlurpParam([1,2,3,4,5,6,7], 8, 9,10; x=1, y=10+2)
+@test s.args == (9,10)
+@test s.kwargs == [(:x => 1), (:y => 12)]
```

Comment:
and this, returned: `Pair[:kwargs => Base.Iterators.Pairs(:x => 1,:y => 12)] == [:x => 1, :y => 12]`

Reply:
```suggestion
@test (s.kwargs[:x] == 1 && s.kwargs[:y] == 12)
```

I'm not sure the ordering is guaranteed, so this is more future-proof.

Reply:
Hmm, 

```jl
using QuickTypes
@qstruct SlurpParam{T}(x::AbstractVector{T}, y=1, args...; kwargs...)
s = SlurpParam([1,2,3,4,5,6,7], 8, 9,10; x=1, y=10+2)

julia> s.kwargs[:x]
ERROR: ArgumentError: invalid index: :x of type Symbol

julia> s.kwargs[1]
:x => 1
```

Should I try changing the `Pair`s

https://github.com/cstjean/QuickTypes.jl/blob/2615ef37b783bc578f12e0cb238238c864e261da/src/QuickTypes.jl#L203-L204

to something like a `Dict`?

Reply:
Yeah, `collect(Pair`) doesn't seem quite right. Can't you just take `$arg_name`?

Basically, we'd like to support function-like behaviour as much as reasonably possible.

```julia
julia> f(; kwargs...) = kwargs[:x]
f (generic function with 1 method)

julia> f(x=2)
2

julia> f(; kwargs...) = kwargs
f (generic function with 1 method)

julia> f(x=2)
pairs(::NamedTuple) with 1 entry:
  :x => 2
```

Reply:
My attempt -> https://github.com/cstjean/QuickTypes.jl/pull/22/commits/14d2ef78fdccd9440afb32cf9f07dd4488f6e7ea

Reply:
Nice!